src/pattern-library/components/patterns/data/DataTablePage.js

```diff
@@ -0,0 +1,434 @@
+              withSource
+            >
+              <div className="w-full h-[250px]">
+                <Scroll>
+                  <DataTable
+                    title="A subset of Nabokov's novels with publish date and original language"
+                    rows={nabokovRows}
+                    columns={nabokovColumns}
+                  />
+                </Scroll>
+              </div>
+            </Library.Demo>
+          </Library.Example>
+        </Library.Pattern>
+
+        <Library.Pattern title="Working with tables">
+          <Library.Example title="Rows, columns and items">
+            <p>
+              <code>DataTable</code> operates on <code>rows</code> and{' '}
+              <code>columns</code>, which are described in detail in their
+              respective Props sections. An <code>item</code> is a single field
+              within a row (an individual table cell).
+            </p>
+            <p>
+              All of the <code>DataTable</code> examples on this page use the
+              same <code>rows</code> data:
+            </p>
+            <div className="h-[300px]">
+              <Next.Code
+                title="rows data used for examples on this page"
+                content={`const rows = [
+    {
+      title: 'Машенька',
+      year: '1926',
+      language: 'Russian',
+      translatedTitle: 'Mary',
+    },
+    {
+      title: 'Король, дама, валет',
+      year: '1928',
+      language: 'Russian',
+      translatedTitle: 'King, Queen, Knave',
+    },
+    {
+      title: 'Защита Лужина',
+      year: '1930',
+      language: 'Russian',
+      translatedTitle: 'The Defense',
+    },
+    {
+      title: 'Соглядатай',
+      year: '1930',
+      language: 'Russian',
+      translatedTitle: 'The Eye',
+    },
+    {
+      title: 'The Real Life of Sebastian Knight',
+      year: '1941',
+      language: 'English',
+    },
+    {
+      title: 'Bend Sinister',
+      year: '1947',
+      language: 'English',
+    },
+    {
+      title: 'Pale Fire',
+      year: '1965',
+      language: 'English',
+    },
+    {
+      title: 'Подвиг',
+      year: '1932',
+      language: 'Russian',
+      translatedTitle: 'Glory',
+    },
+    {
+      title: 'Ada or Ardor: A Family Chronicle',
+      year: '1969',
+      language: 'English',
+    },
+  ];`}
+              />
+            </div>
+          </Library.Example>
+          <Library.Example title="Tables in constrained spaces">
+            <p>
+              <code>DataTable</code> and <code>Table</code> are designed to be
+              compatible with scrolling contexts. The examples on this page are
+              constrained within <code>Scroll</code> components.
+            </p>
+          </Library.Example>
+        </Library.Pattern>
+
+        <Library.Pattern title="Props">
+          <Library.Example title="rows">
+            <p>
+              <code>rows</code> is an Array of basic objects mapping field names
```

Comment:
```suggestion
              <code>rows</code> is an Array of basic objects mapping field names
```

I can't remember if I mentioned this last time around, but there is no hard requirement for objects to be basic/plain as far as I can see. Class instances for example would work fine.